internal/serve/middleware/middleware.go

```diff
@@ -50,7 +51,7 @@ func SignatureMiddleware(signatureVerifier auth.SignatureVerifier) func(next htt
 }
 
 // RecoverHandler is a middleware that recovers from panics and logs the error.
-func RecoverHandler(next http.Handler) http.Handler {
+func RecoverHandler(next http.Handler, appTracker apptracker.AppTracker) http.Handler {
```

Comment:
Here instead of creating a Wrapper, we can add one more layer to the function.

```suggestion
func RecoverHandler(appTracker apptracker.AppTracker) func(http.Handler) http.Handler {
        return func(next http.Handler) http.Handler {
                return http.HandlerFunc(.....
        }
```

Reply:
Done